prototype/mc_shortest_path/arduino_mc_shortest_path/arduino_mc_shortest_path.ino

```diff
@@ -0,0 +1,149 @@
+#include "arduino_serial_port.h"
+#include "monte_carlo.h"
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define BOARD_LENGTH 3
+#define BOARD_SIZE 9 // 3 x 3 board
+
+
+uint8_t totalLength = 0;
+uint8_t initialBoard[9];
+
+void setup() {
+  // Initialize serial
+  Serial.begin(9600);
+}
+
+void loop() {
+  // Receive size of board
+  uint8_t nbValues;
+  readElement(&nbValues, sizeof(nbValues));
+
+  // Receive board
+  int8_t boardValues[nbValues];
+  readArray(nbValues, &boardValues, sizeof(int8_t));
+  uint8_t nPlayers;
+  readElement(&nPlayers, sizeof(nPlayers));
+
+  initialBoard[0] = 1;
+  for (int i = 1; i < nbValues; ++i) {
+    initialBoard[i] = boardValues[i];
+  }
+
+  int minSimulation = 10;
+  int maxSimulation = 48;
+  int targetScore = 10;
+
+  Board board = {boardValues, nPlayers};
+  Game game = {isValidAction,
+               playAction,
+               getScore,
+               getPossibleActions,
+               getNumPossibleActions,
+               removeAction,
+               getBoardSize,
+               isDone,
+               0,
+               1};
+  Action action =
+      mcGame(board, 1, game, minSimulation, maxSimulation, targetScore);
+
+  // Return best action
+  writeElement(&action.xPos, sizeof(uint8_t));
+  writeElement(&action.yPos, sizeof(uint8_t));
+}
+
+uint8_t findCurrentPosition(Board* board) {
+  for (uint8_t i = 0; i < BOARD_SIZE - 1; ++i) {
+    if (board->values[i] == -1) {
+      return i;
+    }
+  }
+  return 0;
+}
+
+void playAction(Board* board, Action* action) {
+  totalLength += board->values[action->xPos * BOARD_LENGTH + action->yPos];
+  uint8_t currentPos = findCurrentPosition(board);
+  board->values[currentPos] = initialBoard[currentPos];
+  board->values[action->xPos * BOARD_LENGTH + action->yPos] = -1;
+}
+
+bool isValidAction(Board* board, Action* action, int player) {
+  uint8_t currentPos = findCurrentPosition(board);
+  uint8_t actionPos = action->xPos * BOARD_LENGTH + action->yPos;
+  if (board->values[actionPos] != -1) {
```

Comment:
It would be more efficient to directly compare currentPos to actionPos as both values will be equal if board->values[actionPos] == -1;